tests/integrations/subprocess/test_Subprocess__stream_output.py

```diff
@@ -97,5 +69,34 @@ def test_stop_func(mock_sub, popen_process, stop_func_ret_val, capsys):
     mock_sub.stream_output(
         "testing", popen_process, stop_func=lambda: stop_func_ret_val
     )
-    assert capsys.readouterr().out == ("output line 1\n" "\n" "output line 3\n")
+    assert capsys.readouterr().out == "output line 1\n" "\n" "output line 3\n"
     mock_sub.cleanup.assert_called_once_with("testing", popen_process)
+
+
+def test_stuck_streamer(mock_sub, popen_process, monkeypatch, capsys):
```

Comment:
definitely open to any feedback for potential improvement

Reply:
Yeah - this is hard to test; this is about as good as it's going to get AFAICT. 

I've pushed a change that mocks `time.time()`; this allows us to shortcut the loop so it doesn't take a full 3 seconds to finish. Ideally, we'd mock `time.sleep()` as well - but if you do that, the parent thread never releases control to the child thread (at least, it doesn't in my testing), so *some* sleeping is required to test the behaviour of the subthread.